resampler/src/speex.rs

```diff
@@ -1409,7 +1402,7 @@ fn resampler_basic_direct_double(
         )))]
         {
             let mut accum: [f64; 4] = [0.0; 4];
-            while j < n {
+            (0..n).step_by(4).for_each(|j| {
```

Comment:
We can iterate over `sinct` and `iptr` using a `zip()` an then sum the result obtained.